rosidl_runtime_py/rosidl_runtime_py/convert.py

```diff
@@ -85,74 +101,107 @@ def to_string(val):
                         val += '...'
             r = str(val)
         else:
-            r = message_to_csv(val, truncate_length)
+            r = message_to_csv(val, truncate_length, no_arr, no_str)
         return r
     result = ''
+    slot_types = get_message_slot_types(msg)
+
     # We rely on __slots__ retaining the order of the fields in the .msg file.
     for field_name in msg.__slots__:
         value = getattr(msg, field_name)
+        field_type = msg._fields_and_field_types[field_name[1:]]
+        has_maximum_size = getattr(slot_types[field_name[1:]], 'has_maximum_size', None)
+
         if result:
             result += ','
-        result += to_string(value)
+
+        result += to_string(value, field_type, has_maximum_size)
     return result
 
 
 # Convert a msg to an OrderedDict. We do this instead of implementing a generic __dict__() method
 # in the msg because we want to preserve order of fields from the .msg file(s).
-def message_to_ordereddict(msg: Any, truncate_length: int = None) -> OrderedDict:
+def message_to_ordereddict(
+        msg: Any,
+        truncate_length: int = None,
+        no_arr: bool = False,
+        no_str: bool = False) -> OrderedDict:
     """
     Convert a ROS message to an OrderedDict.
 
     :param msg: The ROS message to convert.
     :param truncate_length: Truncate values for all message fields to this length.
         This does not truncate the list of fields (ie. the dictionary keys).
+    :param no_arr: Exclude array fields of the message
```

Comment:
@vinnamkim missing period at the end of the sentence.

Reply:
Fixed.